capten/agent/pkg/agent/agent_conf_sso.go

```diff
@@ -0,0 +1,206 @@
+package agent
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"reflect"
+
+	"github.com/intelops/go-common/credentials"
+	"github.com/kube-tarian/kad/capten/agent/pkg/agentpb"
+	"github.com/kube-tarian/kad/capten/agent/pkg/workers"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+func (a *Agent) ConfigureAppSSO(
+	ctx context.Context, req *agentpb.ConfigureAppSSORequest) (*agentpb.ConfigureAppSSOResponse, error) {
+
+	if req.ReleaseName == "" {
+		return &agentpb.ConfigureAppSSOResponse{
+			Status:        agentpb.StatusCode_INVALID_ARGUMENT,
+			StatusMessage: "release name empty",
+		}, nil
+	}
+
+	appConfig, err := a.as.GetAppConfig(req.ReleaseName)
+	if err != nil {
+		return &agentpb.ConfigureAppSSOResponse{
+			Status:        agentpb.StatusCode_INTERNRAL_ERROR,
+			StatusMessage: errors.WithMessage(err, "err fetching appConfig").Error(),
+		}, nil
+	}
+
+	// save client secret in vault
+	if err := saveSsoCredentials(ctx, req.ClientId, req.ClientSecret); err != nil {
+		a.log.Audit("security", "storecred", "failed", "system", "failed to intialize credentails for clientId: %s", req.ClientId)
+		a.log.Errorf("failed to store credentail for ClientId: %s, %v", req.ClientId, err)
```

Comment:
dont log clientid, log the app release name.
"failed to store sso  credential for app"